cisco_aci/datadog_checks/cisco_aci/cisco.py

```diff
@@ -0,0 +1,152 @@
+            service_check_tags.append("url:{}".format(url))
+        service_check_tags.extend(self.check_tags)
+        service_check_tags.extend(instance.get('tags', []))
+
+        try:
+            api.login()
+        except Exception as e:
+            self.log.exception("Cannot login to the Cisco ACI: {}".format(e))
+            self.service_check('cisco_aci.can_connect',
+                               AgentCheck.CRITICAL,
+                               "aci login returned a status of {}".format(e),
+                               tags=service_check_tags)
+            raise
+
+        self.tagger.api = api
+
+        try:
+            tenant = Tenant(self, api, instance, instance_hash)
+            tenant.collect()
+        except Exception as e:
+            self.log.info('tenant collection failed: {}'.format(e))
+            self.service_check('cisco_aci.can_connect',
+                               AgentCheck.CRITICAL,
+                               "aci tenant operations failed, returning a status of {}".format(e),
+                               tags=service_check_tags)
+            api.close()
+            raise
+
+        try:
+            fabric = Fabric(self, api, instance)
+            fabric.collect()
+        except Exception as e:
+            self.log.info('fabric collection failed: {}'.format(e))
+            self.service_check('cisco_aci.can_connect',
+                               AgentCheck.CRITICAL,
+                               "aci fabric operations failed, returning a status of {}".format(e),
+                               tags=service_check_tags)
+            api.close()
+            raise
+
+        try:
+            capacity = Capacity(self, api, instance)
+            capacity.collect()
+        except Exception as e:
+            self.log.info('capacity collection failed: {}'.format(e))
+            self.service_check('cisco_aci.can_connect',
+                               AgentCheck.CRITICAL,
+                               "aci capacity operations failed, returning a status of {}".format(e),
+                               tags=service_check_tags)
+            api.close()
+            raise
+
+        self.service_check('cisco_aci.can_connect',
+                           AgentCheck.OK,
+                           tags=service_check_tags)
+
+        api.close()
+        end = datetime.datetime.now()
+        log_line = "finished running Cisco Check"
+        if _is_affirmative(instance.get('report_timing', False)):
+            log_line += ", took {}".format(end - start)
+        self.log.info(log_line)
+
+    def submit_metrics(self, metrics, tags, instance={}, obj_type="gauge", hostname=None):
+        user_tags = instance.get('tags', [])
+        for mname, mval in metrics.iteritems():
+            tags_to_send = []
+            if mval:
+                if hostname:
+                    tags_to_send += self.check_tags
+                tags_to_send += user_tags
+                tags_to_send += tags
```

Comment:
Let's merge these two tag appends 👍

Reply:
More importantly, it's submitting the wrong tags variable `self.gauge(mname, float(mval), tags=tags, hostname=hostname)`

Reply:
@gmmeyer